R/package.R

```diff
@@ -12,34 +12,39 @@ NULL
-#'   that is added as an extra error to the result object.
-#' @param error_on Whether to throw an error on `R CMD check` failures.
-#'   Note that the check is always completed (unless a timeout happens),
-#'   and the error is only thrown after completion. If `"never"`, then
-#'   no errors are thrown. If `"error"`, then only `ERROR` failures
-#'   generate errors. If `"warning"`, then `WARNING` failures generate
-#'   errors as well. If `"note"`, then any check failure generated an
-#'   error.
-#' @return An S3 object (list) with fields `errors`,
-#'   `warnings` and `notes`. These are all character
-#'   vectors containing the output for the failed check.
+#' @param args Character vector of arguments to pass to `R CMD check`. Pass each
+#'   argument as a single element of this character vector (do not use spaces to
+#'   delimit arguments like you would in the shell). For example, to skip
+#'   running of examples and tests, use `args = c("--no-examples",
+#'   "--no-tests")` and not `args = "--no-examples --no-tests"`. (Note that
+#'   instead of the `--output` option you should use the `check_dir` argument,
+#'   because  `--output` cannot deal with spaces and other special characters on
+#'   Windows.)
+#' @param build_args Character vector of arguments to pass to `R CMD build`.
+#'   Pass each argument as a single element of this character vector (do not use
+#'   spaces to delimit arguments like you would in the shell). For example,
+#'   `build_args = c("--force", "--keep-empty-dirs")` is a correct usage and
+#'   `build_args = "--force --keep-empty-dirs"` is incorrect.
+#' @param check_dir Path to a directory where the check is performed. If this is
+#'   not `NULL`, then the a temporary directory is used, that is cleaned up when
+#'   the returned object is garbage collected.
+#' @param libpath The library path to set for the check. The default uses the
+#'   current library path.
+#' @param repos The `repos` option to set for the check. This is needed for
+#'   cyclic dependency checks if you use the `--as-cran` argument. The default
+#'   uses the current value.
+#' @param timeout Timeout for the check, in seconds, or as a [base::difftime]
+#'   object. If it is not finished before this, it will be killed. `Inf` means
+#'   no timeout. If the check is timed out, that is added as an extra error to
+#'   the result object.
+#' @param error_on Whether to throw an error on `R CMD check` failures. Note
+#'   that the check is always completed (unless a timeout happens), and the
+#'   error is only thrown after completion. If `"never"`, then no errors are
+#'   thrown. If `"error"`, then only `ERROR` failures generate errors. If
+#'   `"warning"`, then `WARNING` failures generate errors as well. If `"note"`,
+#'   then any check failure generated an error.
+#' @return An S3 object (list) with fields `errors`, `warnings` and `notes`.
+#'   These are all character vectors containing the output for the failed check.
```

Comment:
Thanks! Do we actually need all these other changes? It is hard for me to follow what has changed here in addition to `args`. Thanks.

Reply:
The only thing I changed in other files are spelling corrections.
In `package.R`, the only things I changed are `@param args` and `@param build_args`; I felt that both needed the same sort of explanation. It's a little bit long, but I think it's better than the alternative, which was confusing. Feel free to edit/shorten/reject I really don't mind and you probably know best.

Reply:
Thanks! My guess is that your IDE re-formatted the whole roxygen block, unfortunately....

Reply:
Yes that is what happened. I've fixed it now, reverting the other arguments back to how they were before. Is that showning up for you now?

Reply:
Yep, looks great, thanks!